Gitweb:     http://git.kernel.org/linus/214ac9a4ead6cb254451c09d9c8234a76693feb1
Commit:     214ac9a4ead6cb254451c09d9c8234a76693feb1
Parent:     49d7590ce494a971311ca55d8aaa6ea9a87b108f
Author:     Larry Finger <Larry.Finger@lwfinger.net>
AuthorDate: Wed Dec 9 13:25:56 2009 -0600
Committer:  John W. Linville <linville@tuxdriver.com>
CommitDate: Thu Dec 10 16:09:51 2009 -0500

    b43: Remove reset after fatal DMA error
    
    As shown in Kernel Bugzilla #14761, doing a controller restart after a
    fatal DMA error does not accomplish anything other than consume the CPU
    on an affected system. Accordingly, substitute a meaningful message for
    the restart.
    
    Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
    Cc: Stable <stable@vger.kernel.org>        [2.6.32]
    Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 drivers/net/wireless/b43/main.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

--- a/drivers/net/wireless/b43/main.c
+++ b/drivers/net/wireless/b43/main.c
@@ -1784,7 +1784,10 @@ static void b43_do_interrupt_thread(stru
 			       dma_reason[0], dma_reason[1],
 			       dma_reason[2], dma_reason[3],
 			       dma_reason[4], dma_reason[5]);
-			b43_controller_restart(dev, "DMA error");
+			b43err(dev->wl, "This device does not support DMA "
+			       "on your system. Please use PIO instead.\n");
+			b43err(dev->wl, "CONFIG_B43_FORCE_PIO must be set in "
+			       "your kernel configuration.\n");
 			return;
 		}
 		if (merged_dma_reason & B43_DMAIRQ_NONFATALMASK) {
